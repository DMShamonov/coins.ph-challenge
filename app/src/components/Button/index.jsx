// @flow

import * as React from 'react';
import classnames from 'classnames';

import type { PropsType } from './types.js.flow';

/* eslint-disable react/button-has-type */

export default function Button({
  type = 'button',
  disabled,
  className,
  children,
  onClick,
}: PropsType): React.Node {
  return (
    <button
      type={type}
      disabled={disabled}
      className={classnames('button', className)}
      onClick={onClick}
    >
      {children}
    </button>
  );
}
