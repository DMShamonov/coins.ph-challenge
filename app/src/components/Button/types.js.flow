// @flow

import type { Node } from 'react';

export type PropsType = {|
  type?: (
    | 'button'
    | 'reset'
    | 'submit'
  ),
  disabled?: boolean,
  className?: string,
  children?: Node,
  onClick?: Event => void,
|};
