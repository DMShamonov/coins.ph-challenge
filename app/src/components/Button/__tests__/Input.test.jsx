import React from 'react';
import { shallow } from 'enzyme';

import Button from '../index';

describe('`Button` component should render correctly', () => {
  test('with default props', () => {
    expect(shallow(<Button />)).toMatchSnapshot();
  });

  test('when `type` prop is `submit`', () => {
    expect(shallow(<Button type="submit" />)).toMatchSnapshot();
  });

  test('when `disabled` prop is `true`', () => {
    expect(shallow(<Button disabled />)).toMatchSnapshot();
  });

  test('with `className` prop', () => {
    expect(shallow(<Button className="__CLASSNAME__" />)).toMatchSnapshot();
  });

  test('with `children` prop', () => {
    expect(shallow(<Button>__CONTENT__</Button>)).toMatchSnapshot();
  });

  test('with `onClick` prop', () => {
    expect(shallow(<Button onClick={jest.fn()} />)).toMatchSnapshot();
  });
});

describe('Check `Button` props', () => {
  describe('Check callback props', () => {
    test('`onClick` called', () => {
      const onClick = jest.fn();
      const wrapper = shallow(<Button onClick={onClick} />);

      wrapper.simulate('click');

      expect(onClick).toHaveBeenCalled();
    });
  });
});
