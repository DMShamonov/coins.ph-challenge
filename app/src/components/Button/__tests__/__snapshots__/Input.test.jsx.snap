// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`\`Button\` component should render correctly when \`disabled\` prop is \`true\` 1`] = `
<button
  className="button"
  disabled={true}
  type="button"
/>
`;

exports[`\`Button\` component should render correctly when \`type\` prop is \`submit\` 1`] = `
<button
  className="button"
  type="submit"
/>
`;

exports[`\`Button\` component should render correctly with \`children\` prop 1`] = `
<button
  className="button"
  type="button"
>
  __CONTENT__
</button>
`;

exports[`\`Button\` component should render correctly with \`className\` prop 1`] = `
<button
  className="button __CLASSNAME__"
  type="button"
/>
`;

exports[`\`Button\` component should render correctly with \`onClick\` prop 1`] = `
<button
  className="button"
  onClick={[MockFunction]}
  type="button"
/>
`;

exports[`\`Button\` component should render correctly with default props 1`] = `
<button
  className="button"
  type="button"
/>
`;
